From 211d141e719c72a607ba72b1c91d92bcffbfe6b7 Mon Sep 17 00:00:00 2001
From: Tristan Cacqueray <tdecacqu@redhat.com>
Date: Wed, 7 Dec 2016 04:38:25 +0000
Subject: [PATCH] Increase rabbitmq connections attempt

Workers will instantly fail if rabbitmq server is not available. This changes
use connection_attemps and retry_delay pika connection parameters to try to
reconnect before failing
---
 etc/storyboard.conf.sample                     | 6 ++++++
 storyboard/notifications/conf.py               | 4 ++++
 storyboard/notifications/connection_service.py | 4 +++-
 3 files changed, 13 insertions(+), 1 deletion(-)

diff --git a/storyboard/notifications/conf.py b/storyboard/notifications/conf.py
index 393f357..325be8e 100644
--- a/storyboard/notifications/conf.py
+++ b/storyboard/notifications/conf.py
@@ -40,4 +40,8 @@ NOTIFICATION_OPTS = [
     cfg.StrOpt("rabbit_virtual_host", default="/",
                help="The virtual host within which our queues and exchanges "
                     "live."),
+    cfg.IntOpt("rabbit_connection_attempts", default=6,
+               help="The number of connection attempts before giving-up"),
+    cfg.IntOpt("rabbit_retry_delay", default=10,
+               help="The interval between connection attempts (in seconds)")
 ]
diff --git a/storyboard/notifications/connection_service.py b/storyboard/notifications/connection_service.py
index d84a55e..90efeea 100644
--- a/storyboard/notifications/connection_service.py
+++ b/storyboard/notifications/connection_service.py
@@ -56,7 +56,9 @@ class ConnectionService(object):
             conf.rabbit_host,
             conf.rabbit_port,
             conf.rabbit_virtual_host,
-            self._connection_credentials)
+            self._connection_credentials,
+            connection_attempts=conf.rabbit_connection_attempts,
+            retry_delay=conf.rabbit_retry_delay)
 
     def _connect(self):
         """This method connects to RabbitMQ, establishes a channel, declares
-- 
2.7.3

